samza-api/src/main/java/org/apache/samza/context/ApplicationDefinedContainerContext.java

```diff
@@ -0,0 +1,22 @@
+package org.apache.samza.context;
+
+/**
+ * An application should implement this to contain any runtime objects required to initialize the processing logic.
+ * <p>
+ * A single instance of this will be created in each container, so it will be shared between all of the tasks on a
+ * single container.
```

Comment:
Add a comment here that the context will be recreated if the container model changes (e.g. during rebalancing) (suggestion by @shanthoosh).